core/src/main/java/org/infinispan/stats/impl/ClusterCacheStatsImpl.java

```diff
@@ -478,6 +486,9 @@ private DistributedCacheStatsCallable(String cacheName) {
 
       @Override
       public Map<String, Number> apply(EmbeddedCacheManager embeddedCacheManager) {
+         if (!embeddedCacheManager.cacheExists(cacheName))
```

Comment:
I know we discussed this in the chat @ryanemerson but I'm having second thoughts: this check won't help if the cache/manager is up during `cacheExists(cacheName)` but shuts down immediately after, so now I'm leaning towards checking for `CacheConfigurationException` and `IllegalLifecycleStateException` in the `TriConsumer` lambda (https://github.com/infinispan/infinispan/pull/9300/files#diff-229b5557c0fd98e85c2f7d4dfc489feac26e1d15fa3e0b337112598975b35d89R110).

I know `CacheConfigurationException` can mean other things too (e.g. the cache didn't start because an internal component didn't like the configuration, but maybe that's fine.

Reply:
Ok, I think that makes sense. Any thoughts on how we should handle these exceptions?

Simply a debug message? WARN or INFO seems excessive, as I consider it correct behaviour for stats not to be computed on a node that does not contain a cache definition.

```
            if (t instanceof CacheConfigurationException || t instanceof IllegalLifecycleStateException) {
               log.debugf(t,"Exception encountered on %s", a);
               return;
            }
```

Reply:
I was thinking trace/none :)

If you do add a log, you should include the cache name in the message.